src/app/modules/item/components/permissions-edit-dialog/permissions-edit-dialog.component.ts

```diff
@@ -0,0 +1,146 @@
+    },
+    {
+      value: 'content',
+      label: 'Content',
+      comment: 'User(s) can see the content of this item'
+    },
+    {
+      value: 'content_with_descendants',
+      label: 'Content and descendants',
+      comment: 'User(s) can also see the content of this items descendants (when possible for this group)'
+    },
+    {
+      value: 'solution',
+      label: 'Solution',
+      comment: 'User(s) can also see the solution of this items and its descendants (when possible for this group)'
+    }
+  ];
+
+  readonly canGrantViewValues = [
+    {
+      value: 'none',
+      label: 'Nothing',
+      comment: 'User(s) can\'t grant any access to this item'
+    },
+    {
+      value: 'enter',
+      label: 'Info & enter',
+      comment: 'User(s) can grant "Can view: info" and  "Can enter" access'
+    },
+    {
+      value: 'content',
+      label: 'Content',
+      comment: 'User(s) can also grant "Can view: content" access'
+    },
+    {
+      value: 'content_with_descendants',
+      label: 'Content and descendants',
+      comment: 'User(s) can also grant "Can view: content and descendants" access'
+    },
+    {
+      value: 'solution',
+      label: 'Solution',
+      comment: 'User(s) can also grant "Can view: solution" access',
+      disabled: true
+    },
+    {
+      value: 'solution_with_grant',
+      label: 'Solution and grant',
+      comment: 'User(s) can also grant "Can grant view" access',
+      disabled: true
+    }
+  ];
+
+  readonly canWatchValues = [
+    {
+      value: 'none',
+      label: 'Nothing',
+      comment: 'User(s) can\'t watch the activity of others on this item'
+    },
+    {
+      value: 'result',
+      label: 'Result',
+      comment: 'User(s) can view information about submissions and scores of others on this item, but not their answers'
+    },
+    {
+      value: 'answer',
+      label: 'Answer',
+      comment: 'User(s) can also look at other people\'s answers on this item'
+    },
+    {
+      value: 'answer_with_grant',
+      label: 'Answer and grant',
+      comment: 'User(s) can also grant "Can watch" access to others'
+    }
+  ];
+
+  readonly canEditValues = [
+    {
+      value: 'none',
+      label: 'Nothing',
+      comment: 'User(s) can\'t make any changes to this item'
+    },
+    {
+      value: 'children',
+      label: 'Children',
+      comment: 'User(s) can add children to this item and edit how permissions propagate to them'
+    },
+    {
+      value: 'all',
+      label: 'All',
+      comment: 'User(s) can also edit the content of the item itself, but may not delete it'
+    },
+    {
+      value: 'all_with_grant',
+      label: 'All and grant',
+      comment: 'User(s) can also give "Can edit" access to others'
+    }
+  ];
+
+  constructor(
+    public dialogRef: MatDialogRef<PermissionsEditDialogComponent>,
+    public dialog: MatDialog,
+    @Inject(MAT_DIALOG_DATA) public data: PermissionsEditData,
```

Comment:
I do not have the full view of its usage, but this seems to be a way of sharing data between components. I don't think that's a great idea. If there is an actual reason to do it this way, we should discuss it first. Please check what it is used for.

Reply:
I use the Primeng dialog component, so this wasn't needed anymore.